test/token/ERC20/extensions/ERC4626.test.js

```diff
@@ -44,6 +45,81 @@ contract('ERC4626', function (accounts) {
     }
   });
 
+  describe('reentrancy', async function () {
+    const reenterType = {
+      Before: 1,
+      After: 2,
+    };
+
+    let amount;
+    let token;
+    let vault;
+
+    beforeEach(async function () {
+      amount = web3.utils.toBN(10).pow(decimals);
+      token = await ERC4626ReentrantAsset.new();
+
+      vault = await ERC4626.new('', '', token.address);
+      await token.$_mint(holder, amount);
+      await token.$_mint(token.address, amount); // Requires balance to reenter
+    });
+
+    // During a `_deposit`, the vault does `transferFrom(depositor, vault, assets)` -> `_mint(receiver, shares)`
+    // such that a reentrancy BEFORE the transfer guarantees the price is kept the same.
+    // If the order of transfer -> mint is changed to mint -> transfer, the reentrancy could be triggered on an
+    // intermediate state in which the ratio of assets/shares has been decreased (more shares than assets).
+    it('reenters before deposit without changing the price', async function () {
+      // Schedules a reentrancy from the token contract
+      const reenterAmount = 1;
+      await token.functionCall(token.address, vault.contract.methods.approve(vault.address, reenterAmount).encodeABI());
```

Comment:
You are targetting `token.address` but you are encoding using `vault.contract.methods` ?

IMO that would be simpler (if it is what you are trying to do)
```suggestion
      await token.$_approve(token.address, vault.address, reenterAmount);
```